src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImpl.java

```diff
@@ -219,6 +219,13 @@ else if (caseEvent.getCategory().equals(CategoryDTO.CategoryName.DISABLE_RESPOND
         }
       }
 
+      else if (caseEvent.getCategory().equals(CategoryDTO.CategoryName.REPLACED) && category.getNewCaseSampleUnitType().equals(SampleUnitType.B))  {
+          CaseGroup caseGroup = caseGroupRepo.findById(targetCase.getCaseGroupId());
```

Comment:
This variable is only used in the line below; I'd suggest getting rid of it and just going `newCase.setPartyId((caseGroupRepo.findById(targetCase.getCaseGroupId)).getPartyId());`